cpp/daal/src/algorithms/kernel_function/kernel_function_rbf_csr_fast_impl.i

```diff
@@ -168,7 +159,17 @@ services::Status KernelImplRBF<fastCSR, algorithmFPType, cpu>::computeInternalMa
 
     if (a1 == a2)
     {
-        SpBlas<algorithmFPType, cpu>::xsyrk_a_at(dataA1, colIndicesA1, rowOffsetsA1, nVectors1, a1->getNumberOfColumns(), dataR);
+        ReadRowsCSR<algorithmFPType, cpu> mtA1(dynamic_cast<CSRNumericTableIface *>(const_cast<NumericTable *>(a1)), 0, nVectors1);
+        DAAL_CHECK_BLOCK_STATUS(mtA1);
+        const algorithmFPType * dataA1 = mtA1.values();
+        const size_t * colIndicesA1    = mtA1.cols();
+        const size_t * rowOffsetsA1    = mtA1.rows();
+
+        WriteOnlyRows<algorithmFPType, cpu> mtR(r, 0, nVectors1);
+        DAAL_CHECK_BLOCK_STATUS(mtR);
+        algorithmFPType * dataR = mtR.get();
+
+        SpBlas<algorithmFPType, cpu>::xsyrk_a_at(dataA1, colIndicesA1, rowOffsetsA1, nVectors1, a1->getNumberOfColumns(), dataR, nVectors2);
 
         daal::threader_for_optional(nVectors1, nVectors1, [=](size_t i) {
```

Comment:
Why optional not just simple parallel for?

Reply:
this is a case for smo, some degradation is obtained in this case for a multiclass. We don't care too much about the performance of this case.

Reply:
If you don't care about perf of the case - better to make it simpler :)

Reply:
Sure, but degradation for SMO is bad :(